borg/archive.py

```diff
@@ -786,11 +791,11 @@ def add_callback(chunk):
 
         def add_reference(id_, size, csize, cdata=None):
             try:
-                count, _, _ = self.chunks[id_]
-                self.chunks[id_] = count + 1, size, csize
```

Comment:
why not `chunk = chunks.get(id_, ChunkIndexEntry(refcount=0, size=size, csize=csize))`
followed by a `chunks[id_] = chunk.replace(refcount=chunk.refcount+1)`

would putting the assertion into the repair branch make sense as well?